### PR TITLE
GPTJ shape issues

### PR DESCRIPTION
**Pull request by Vimalakirti**
Moved the shape fixes from #109

### CODE REVIEW

File: src/layer/constantofshape.rs [hunk 1]
```diff
@@ -19,7 +19,7 @@ impl Layer for ConstOfShapeLayer {
   ) -> (Graph, Vec<Vec<usize>>, Vec<DatumType>) {
     let mut graph = Graph::new();
 
-    let value = Fr::from(attributes.iter().filter(|x| x.name == "value").next().unwrap().i);
+    let value = Fr::from(attributes.iter().filter(|x| x.name == "value").next().unwrap().t.clone().unwrap().raw_data[0]);
```
Comment (liliatangxy) [inline src/layer/constantofshape.rs]: will need to change the layer's output type to be based on the tensor's type

Comment (Vimalakirti) [reply src/layer/constantofshape.rs]: added, please check

**Review by liliatangxy — DISMISSED** [review]

File: src/layer/pow.rs [hunk 2]
```diff
@@ -16,7 +41,34 @@ impl Layer for PowLayer {
     _attributes: &Vec<&AttributeProto>,
   ) -> (Graph, Vec<Vec<usize>>, Vec<DatumType>) {
     let mut graph = Graph::new();
-    assert!(constants[1].unwrap().0.first().unwrap() == &Fr::from(2 * *onnx::SF as u32));
+
+    // if both constants[0] and constants[1] are Some
```
Comment (ddkang) [inline src/layer/pow.rs:54]: Not clear why you're doing this

Comment (liliatangxy) [reply src/layer/pow.rs:54]: can comment something like if both operands are constants then we can precompute/omit proof

Comment (Vimalakirti) [reply src/layer/pow.rs:54]: Added comments to explain. (Also included a TODO tag in comments)

Comment (ddkang) [reply src/layer/pow.rs:54]: Is this insecure as is? If so, add a runtime warning.

Comment (Vimalakirti) [reply src/layer/pow.rs:54]: It is secure because both inputs are constants during graph compilation.

**Review by ddkang — DISMISSED** [review]

**Comment by Vimalakirti** [comment]
The new commit included the support for dim-0 Add and Sub. (This is also a fix I used for the GPTj job on CC, but I forgot to include it in this PR.)

**Comment by Vimalakirti** [comment]
I want to double check if we need to consider datumType in the constant of Pow.
That's the reason why this PR is turned to a draft.

**Review by liliatangxy — APPROVED** [review]

**Review by ddkang — APPROVED** [review]

**Pull request merged**